support/k8s/k8s-workload-registrar/controller.go

```diff
@@ -210,6 +220,44 @@ func (c *Controller) makeID(pathFmt string, pathArgs ...interface{}) string {
 	return id.String()
 }
 
+// Ensure that there is exactly one entry matching these selectors, and it's this one
+func (c *Controller) syncEntry(ctx context.Context, entry *common.RegistrationEntry) error {
+	entries, err := c.c.R.ListBySelectors(ctx, &common.Selectors{
+		Entries: entry.Selectors,
+	})
+	if err != nil {
+		return errs.New("unable to list by pod entries: %v", err)
+	}
+
+	var existing *common.RegistrationEntry = nil
```

Comment:
nit: it is idiomatic to leave off the initializer if initializing to the zero value.

Reply:
Done